src/mlir/generator.cc

```diff
@@ -480,19 +447,13 @@ namespace mlir::verona
         "Return operation without parent function", getLocation(ast));
     }
 
+    // Either retrurns empty or auto-cast'ed value
     if (expr.hasValue())
     {
       assert(retTy && "Return value from a void function");
       // Emit the cast if necessary (we trust the ast)
-      if (expr.get().getType() != retTy)
-      {
-        auto op =
-          genOperation(expr.get().getLoc(), "verona.cast", {expr.get()}, retTy);
-        if (auto err = op.takeError())
-          return std::move(err);
```

Comment:
The error handling code here is removed by the refactor.

Reply:
This is intentional. The code could never return an error.

`llvm::Expected<>` and `llvm::Error` assert if the errors are not treated, so there's no way we can use a value that *could* error without handling it.